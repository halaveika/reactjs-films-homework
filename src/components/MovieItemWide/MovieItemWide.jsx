import React from 'react';
import { useNavigate } from 'react-router-dom';
import { Card, Button } from 'antd';
import { CaretRightOutlined } from '@ant-design/icons';
import PropTypes from 'prop-types';
import MovieInfo from '../MovieInfo';
import WatchVideoModal from '../WatchVideoModal';
import './MovieItemWide.scss';

export default function MovieItemWide({
  id, title, genres, vote_average, poster, overview, getDetails,
}) {
  const navigate = useNavigate();

  const handleDetailPageNavigation = async () => {
    await getDetails(id);
    navigate('/details');
  };

  return (
    <div className="movie-item-wide-container">
      <div className="overlay" onClick={handleDetailPageNavigation} />
      <Card className="movie-item-wide">
        <img className="poster" alt="Poster!" src={poster} />
        <WatchVideoModal id={id}>
          <Button className="watch-btn" type="primary" shape="circle">
            <CaretRightOutlined
              className="icon"
              style={{ fontSize: '250%' }}
            />
          </Button>
        </WatchVideoModal>
        <MovieInfo
          active={false}
          className="movie-info__active"
          title={title}
          genres={genres}
          vote_average={vote_average}
        />

      </Card>
      <span className="overview">{overview}</span>
    </div>

  );
}

MovieItemWide.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string,
  genres: PropTypes.arrayOf(PropTypes.string),
  vote_average: PropTypes.number,
  poster: PropTypes.string,
  overview: PropTypes.string,
  getDetails: PropTypes.func.isRequired,
};
