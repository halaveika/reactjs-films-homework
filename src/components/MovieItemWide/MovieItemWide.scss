.overlay {
  position: absolute;
  display: none;
  width: 200px;
  height: 250px;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.7);
  z-index: 2;
  cursor: pointer;
}

.movie-item-wide-container {
  position: relative;
  display: flex;
  flex-direction: row;
  align-content: flex-start;
  width: 100%;
  height: auto;
  padding: 0;
  margin: 30px;

  & .watch-btn {
    display: none;
  }

  &:hover .overlay {
    display: block;
  }

  &:hover .watch-btn {
    display: block;
    position: absolute;
    width: 50px;
    height: 50px;
    left: 75px;
    top: 50px;
    z-index: 10;
    cursor: pointer;

    &::after {
      content: 'Watch Now';
      position: absolute;
      top: 80px;
      left: -20px;
      font-size: 16px;
      color: #fff;
    }

    & .icon {
      position: absolute;
      top: 7px;
      left: 8px;
    }
  }

  & .overview {
    display: block;
    position: relative;
    width: auto;
    height: 100%;
    font-size: 18px;
    color: rgba(0, 0, 0, 0.85);
    top: 0;
    left: 0;
  }
}

.movie-item-wide {
  position: relative;
  width: 200px;
  height: 305px;
  padding: 0;
  margin-right: 30px;
}

.poster {
  width: 200px;
  height: 250px;

  &__active {
    width: 200px;
    height: 250px;
  }
}
