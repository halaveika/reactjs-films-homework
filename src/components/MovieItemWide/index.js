import MovieItemWide from './MovieItemWide';

export default MovieItemWide;
