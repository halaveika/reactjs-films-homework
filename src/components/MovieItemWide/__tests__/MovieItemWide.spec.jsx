import React from 'react';
import { create, act } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import MovieItemWide from '../MovieItemWide';

const mockMovieItemProps = {
  id: 2412412,
  title: 'Fight club',
  genres: ['drama', 'sport', 'comedy'],
  vote_average: 8,
  poster: 'image_path',
  overview: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
  getDetails: jest.fn(),
};

jest.spyOn(window, 'scrollTo').mockImplementation();

describe('test MovieItemWide component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  const component = create(
    <BrowserRouter>
      <MovieItemWide
        id={mockMovieItemProps.id}
        title={mockMovieItemProps.title}
        genres={mockMovieItemProps.genres}
        vote_average={mockMovieItemProps.vote_average}
        poster={mockMovieItemProps.poster}
        overview={mockMovieItemProps.overview}
        getDetails={mockMovieItemProps.getDetails}
      />
    </BrowserRouter>
    ,
  );

  it('should render MovieItemWide component', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handleDetailPageNavigation should change state', async () => {
    const { root } = component;
    const clickable = root.findByProps({ className: 'overlay' });
    await act(async () => { clickable.props.onClick(); });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
