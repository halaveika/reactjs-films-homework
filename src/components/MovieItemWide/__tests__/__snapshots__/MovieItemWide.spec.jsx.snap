// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MovieItemWide component handleDetailPageNavigation should change state 1`] = `
<div
  className="movie-item-wide-container"
>
  <div
    className="overlay"
    onClick={[Function]}
  />
  <div
    className="ant-card ant-card-bordered movie-item-wide"
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <img
        alt="Poster!"
        className="poster"
        src="image_path"
      />
      <button
        className="ant-btn ant-btn-primary ant-btn-circle watch-btn"
        onClick={[Function]}
        type="button"
      >
        <span
          aria-label="caret-right"
          className="anticon anticon-caret-right icon"
          role="img"
          style={
            Object {
              "fontSize": "250%",
            }
          }
        >
          <svg
            aria-hidden="true"
            data-icon="caret-right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M715.8 493.5L335 165.1c-14.2-12.2-35-1.2-35 18.5v656.8c0 19.7 20.8 30.7 35 18.5l380.8-328.4c10.9-9.4 10.9-27.6 0-37z"
            />
          </svg>
        </span>
      </button>
      <div
        className="movie-item-container"
      >
        <span
          className="movie-item_title"
        >
          Fight club
        </span>
        <span
          className="movie-item_vote-average"
        >
          8
        </span>
        <span
          className="movie-item_genre"
        >
          drama, sport, comedy
        </span>
      </div>
    </div>
  </div>
  <span
    className="overview"
  >
    Lorem Ipsum is simply dummy text of the printing and typesetting industry.
  </span>
</div>
`;

exports[`test MovieItemWide component should render MovieItemWide component 1`] = `
<div
  className="movie-item-wide-container"
>
  <div
    className="overlay"
    onClick={[Function]}
  />
  <div
    className="ant-card ant-card-bordered movie-item-wide"
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <img
        alt="Poster!"
        className="poster"
        src="image_path"
      />
      <button
        className="ant-btn ant-btn-primary ant-btn-circle watch-btn"
        onClick={[Function]}
        type="button"
      >
        <span
          aria-label="caret-right"
          className="anticon anticon-caret-right icon"
          role="img"
          style={
            Object {
              "fontSize": "250%",
            }
          }
        >
          <svg
            aria-hidden="true"
            data-icon="caret-right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M715.8 493.5L335 165.1c-14.2-12.2-35-1.2-35 18.5v656.8c0 19.7 20.8 30.7 35 18.5l380.8-328.4c10.9-9.4 10.9-27.6 0-37z"
            />
          </svg>
        </span>
      </button>
      <div
        className="movie-item-container"
      >
        <span
          className="movie-item_title"
        >
          Fight club
        </span>
        <span
          className="movie-item_vote-average"
        >
          8
        </span>
        <span
          className="movie-item_genre"
        >
          drama, sport, comedy
        </span>
      </div>
    </div>
  </div>
  <span
    className="overview"
  >
    Lorem Ipsum is simply dummy text of the printing and typesetting industry.
  </span>
</div>
`;
